src/main/java/hudson/remoting/Channel.java

```diff
@@ -159,7 +159,7 @@
      * Requests that are sent to the remote side for execution, yet we are waiting locally until
      * we hear back their responses.
      */
-    /*package*/ final Map<Integer,Request<? extends Serializable,? extends Throwable>> pendingCalls = new Hashtable<>();
+    /*package*/ final Map<Integer,Request<? extends Serializable,? extends Throwable>> pendingCalls = new ConcurrentHashMap<>();
```

Comment:
ConcurrentHashMap is a good model for what this needs to do. Based upon the existing implementation, it's possible it doesn't need thread safety but that seems unlikely. ConcurrentHashMap has low negative impact, but still provides thread safety if it really is needed.